packages/sit-onyx/src/components/OnyxAppLayout/OnyxAppLayout.vue

```diff
@@ -0,0 +1,84 @@
+<script lang="ts" setup>
+import { useSlots } from "vue";
+
+const props = withDefaults(
+  defineProps<{
+    /** Whether the nav bar will stick to the left or to the top of the app */
+    navBarAlignment: "top" | "left";
+  }>(),
+  { navBarAlignment: "top" },
+);
+defineSlots<{
+  /** Navigation area of the application */
+  navBar(props: Record<string, never>): unknown;
+  /** Page content area of the application */
+  default(props: Record<string, never>): unknown;
+  /** Overlays that cover the page and exclude the nav area */
+  pageOverlay(props: Record<string, never>): unknown;
+  /** Overlays that cover the complete page */
+  appOverlay(props: Record<string, never>): unknown;
+}>();
+
+const slots = useSlots();
+</script>
+
+<template>
+  <div class="onyx-app" :class="{ 'onyx-app--horizontal': props.navBarAlignment === 'left' }">
+    <nav class="onyx-app__nav">
+      <slot name="navBar"></slot>
+    </nav>
+    <div class="onyx-app__page">
+      <slot></slot>
+    </div>
+    <div v-if="slots.pageOverlay" class="onyx-app__page-overlay">
+      <slot name="pageOverlay"></slot>
+    </div>
+    <div v-if="slots.appOverlay" class="onyx-app__app-overlay">
+      <slot name="appOverlay"></slot>
+    </div>
+  </div>
+</template>
+
+<style lang="scss">
+.onyx-app {
+  height: 100vh;
+  width: 100vw;
+  display: grid;
+  grid-template-rows: max-content 1fr;
+  grid-template-columns: 1fr;
+  grid-template-areas:
+    "nav"
+    "page";
+
+  &--horizontal {
+    grid-template-rows: 1fr;
+    grid-template-columns: max-content 1fr;
+    grid-template-areas: "nav page";
+  }
+
+  &__nav {
+    grid-area: nav;
+    z-index: var(--onyx-z-index-navigation);
+  }
+  &__page {
+    grid-area: page;
+    overflow: hidden auto;
+    position: relative;
+  }
+  &__page-overlay {
+    grid-area: page;
+    z-index: var(--onyx-z-index-page-overlay);
+  }
+  &__app-overlay {
+    grid-column: 1 / -1;
+    grid-row: 1 / -1;
+    top: 0;
+    left: 0;
+    bottom: 0;
+    right: 0;
+    z-index: var(--onyx-z-index-app-overlay);
+    position: absolute;
+    background-color: rgba(0, 0, 0, 0.5);
```

Comment:
I don't think we should set styles other than layout for layout components

Reply:
which one do you mean? the backdrop color?

Reply:
or all the background-colors?

Reply:
Background colors in general...but I am not 100% sure, maybe we can also ask the others.
Example: What if a project wants to have a transparent header, adjust the overlay transparency because they need the content to be visible behind it

Reply:
Personally I think a layout component (in the sense of atomic design) is only allowed to to layout stuff like flex, grid, positions etc. but no visuals like colors

Reply:
Fine for me to keep the colors for now and create a follow up ticket to discuss it